packages/remix/src/build.ts

```diff
@@ -149,17 +150,37 @@ export const build: BuildV2 = async ({
       join(DEFAULTS_PATH, 'entry.server.jsx'),
       join(appDirectory, 'entry.server.jsx')
     );
-    if (!pkg.dependencies['@vercel/remix-entry-server']) {
+    if (!pkg.dependencies['@vercel/remix']) {
+      // Dependency version resolution logic
+      // 1. Users app is on 1.9.0 -> we install the 1.10.0 (minimum) version of our fork (`@vercel/remix`)
+      // 2. Users app is on 1.11.0 (a version greater than 1.10.0 and less than the latest version of the fork) -> we install the (matching) 1.11.0 version of `@vercel/remix`
+      // 3. Users app is on something greater than our latest version of the fork -> we install the latest version of our fork
+
+      // remixVersion is the version of the `@remix-run/dev` package in the *users' app*
+      const usersRemixVersion = semver.gt(remixVersion, '1.10.0')
+        ? remixVersion
+        : '1.10.0'; // our minimum supported version is 1.10.0
+
       // Prevent frozen lockfile rejections
       const envForAddDep = { ...spawnOpts.env };
       delete envForAddDep.CI;
       delete envForAddDep.VERCEL;
       delete envForAddDep.NOW_BUILDER;
-      await addDependency(cliType, ['@vercel/remix-entry-server'], {
-        ...spawnOpts,
-        env: envForAddDep,
-        cwd: entrypointFsDirname,
-      });
+      await addDependency(
+        cliType,
+        [
+          `@vercel/remix@${
+            usersRemixVersion > require('@remix-run/dev').version
```

Comment:
Can't use `>` on semver string 😄

Reply:
Oh, I guess you can cause it'll coerce. Still kinda surprising though, maybe we should use `semver.gt()` here too.